reporting/report_test.go

```diff
@@ -41,3 +44,16 @@ JPcHeO7M6FohKgcEHX84koQDN98J/L7pFlSoU7WOl6f8BKavIdeSTPS9qQYWdQuT
 		t.Errorf("got %+v want %+v", key, results)
 	}
 }
+
+func TestWriteReportInNonExistingDir(t *testing.T) {
+	report := Init()
+
+	tempDir := os.TempDir()
+	resultFile := filepath.Join(tempDir, "test_temp_dir", "sub_dir", "report.yaml")
+	err := report.WriteFile([]string{resultFile}, &config.Config{Name: "report", Version: "5"})
+	if err != nil {
+		t.Error(err)
+	}
+
+	os.RemoveAll(filepath.Join(tempDir, "test_temp_dir"))
```

Comment:
Question: is the `os.TempDir()` return the main temp dir of the system, or create a new dir inside that system temp dir?

Because if it is the second option, we want to clear the dir returned by the `os.TempDir`.

Reply:
`os.TempDir()` returns the OS temp directory, it doesn't create anything.

Reply:
I'm not entirely sure this is the right course of action - on some systems the temp dir might not even be writable for the test. But it did work on mine, and if it passes CI I guess that's good enough.